feature-libs/customer-ticketing/components/customer-ticketing-close/customer-ticketing-close.component.ts

```diff
@@ -1,17 +1,56 @@
-import { Component } from '@angular/core';
+import {
+  Component,
+  ElementRef,
+  ViewChild,
+  ViewContainerRef,
+} from '@angular/core';
 import { STATUS } from '@spartacus/customer-ticketing/root';
-import { Observable } from 'rxjs';
-import { map } from 'rxjs/operators';
-import { CustomerTicketingService } from '../customer-ticketing.service';
+import { LaunchDialogService, LAUNCH_CALLER } from '@spartacus/storefront';
+import { combineLatest, Observable, Subscription } from 'rxjs';
+import { map, take } from 'rxjs/operators';
+import { CustomerTicketingDetailsService } from '../customer-ticketing-details.service';
 
 @Component({
   selector: 'cx-customer-ticketing-close',
   templateUrl: './customer-ticketing-close.component.html',
 })
 export class CustomerTicketingCloseComponent {
-  constructor(protected customerTicketingService: CustomerTicketingService) {}
+  protected subscription = new Subscription();
 
-  isStatusOpen$: Observable<boolean> = this.customerTicketingService
-    .getTicketStatus()
-    .pipe(map((status) => status === STATUS.OPEN));
+  @ViewChild('element') element: ElementRef;
+
+  enableCloseButton$: Observable<boolean | undefined> = combineLatest([
+    this.customerTicketingDetailsService.getTicketStatus(),
+    this.customerTicketingDetailsService.getAvailableTransitionStatus(),
+  ]).pipe(
+    map(
+      ([ticketStatus, availableStatus]) =>
+        (ticketStatus === STATUS.OPEN || ticketStatus === STATUS.INPROCESS) &&
+        availableStatus?.some(
+          (status) => status.id.toUpperCase() === STATUS.CLOSE
+        )
+    )
+  );
+
+  constructor(
+    protected customerTicketingDetailsService: CustomerTicketingDetailsService,
+    protected launchDialogService: LaunchDialogService,
+    protected vcr: ViewContainerRef
+  ) {}
+
+  openDialog() {
+    const dialog = this.launchDialogService.openDialog(
+      LAUNCH_CALLER.CUSTOMER_TICKETING_CLOSE,
+      this.element,
+      this.vcr
+    );
+
+    if (dialog) {
+      this.subscription.add(dialog.pipe(take(1)).subscribe());
+    }
+  }
+
+  ngOnDestroy(): void {
```

Comment:
there's a validation failing here, maybe trying extending OnDestroy in the class